fog/overseer/server/src/worker.rs

```diff
@@ -0,0 +1,511 @@
+        &self,
+    ) -> Result<Vec<CompressedRistrettoPublic>, OverseerError> {
+        // An outanding key is one that Fog Ingest is still obligated to be
+        // scanning blocks with on behalf of users.
+        let outstanding_keys_filters = IngressPublicKeyRecordFilters {
+            // A lost key can never be outstanding because it will never again
+            // be used to scan blocks.
+            should_include_lost_keys: false,
+            // Its possible for a retired key to be outstanding if its public
+            // expiry is greater than its last scanned block, so we have to
+            // include retired keys in this query.
+            should_include_retired_keys: true,
+            // If a key has expired- i.e. its last scanned block is greater
+            // than or equal to its public expiry- then it will no longer scan
+            // blocks. Therefore, we need to include unexpired keys because they
+            // are still supposed to be scanned by Fog.
+            should_only_include_unexpired_keys: true,
+        };
+
+        // First, find the "inactive_outstanding_keys" which are outstanding
+        // keys that we've grabbed from the RecoveryDb.
+        //
+        // TODO: Add a config that allows us to set this start block.
+        let ingress_public_key_records: Vec<IngressPublicKeyRecord> =
+            self.recovery_db.get_ingress_key_records(
+                /* start_block_at_least= */ 0,
+                outstanding_keys_filters,
+            )?;
+
+        Ok(ingress_public_key_records
+            .iter()
+            .map(|record| record.key)
+            .collect())
+    }
+
+    /// Tries to report a lost ingress key.
+    fn report_lost_ingress_key(
+        &self,
+        inactive_outstanding_key: CompressedRistrettoPublic,
+    ) -> Result<(), OverseerError> {
+        let result = retry_with_index(
+            Fixed::from_millis(200).take(Self::NUMBER_OF_TRIES.try_into().unwrap()),
+            |current_try| match self
+                .recovery_db
+                .report_lost_ingress_key(inactive_outstanding_key)
+            {
+                Ok(_) => {
+                    log::info!(
+                        self.logger,
+                        "The following key was successfully reported as lost: {}",
+                        inactive_outstanding_key
+                    );
+                    OperationResult::Ok(())
+                }
+                Err(_) => {
+                    let number_of_remaining_tries = Self::NUMBER_OF_TRIES - current_try;
+                    let error_message = match number_of_remaining_tries {
+                        0 => format!("Did not succeed in reporting lost ingress key {} within {} tries.", inactive_outstanding_key, Self::NUMBER_OF_TRIES),
+                        _ => format!("The following key  was not successfully reported as lost: {}. Will try {} more times", inactive_outstanding_key, number_of_remaining_tries),
```

Comment:
```suggestion
                        _ => format!("The following key was not successfully reported as lost: {}. Will try {} more times", inactive_outstanding_key, number_of_remaining_tries),
```

Reply:
Might be worth to also include the error in the formatting.

Reply:
Done. Added underlying error message to retry error messages.